app/services/paypal/refresh_identity.rb

```diff
@@ -0,0 +1,53 @@
+class Paypal::RefreshIdentity < BaseService
+  def initialize(publisher_id:)
+    @publisher = Publisher.find(publisher_id)
+    @paypal_connection = @publisher.paypal_connection
+  end
+
+  def perform
+    return if @paypal_connection.nil? || @paypal_connection.refresh_token.nil?
+    begin
+      access_token = fetch_access_token
+      raise_token_exception if access_token.nil?
+    rescue => e
+      require 'sentry-raven'
+      Raven.capture_exception(e)
+    end
+
+    user_info_response = Faraday.get("#{Rails.application.secrets[:paypal_api_uri]}/v1/identity/oauth2/userinfo") do |req|
+      req.headers['Authorization'] = "Bearer #{access_token}"
+      req.headers['Accept'] = 'application/json'
+      req.params['schema'] = "paypalv1.1"
+    end
+
+    user_info = JSON.parse(user_info_response.body)
+    paypal_connection = PaypalConnection.find_by(
+      user_id: @publisher.id,
+      hidden: false
+    )
+    paypal_connection.update(
+      email: user_info['emails'][0]['value'],
```

Comment:
Let's use `.dig` for accessing values in this class similarly to the controller.